src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackageCommand.cs

```diff
@@ -0,0 +1,162 @@
+                    CommandOptionType.SingleValue);
+
+                listpkg.OnExecute(async () =>
+                {
+                    var logger = getLogger();
+
+                    var settings = ProcessConfigFile(config.Value(), path.Value);
+                    var sources = ParseArgumentWithCommas(source.Value());
+                    
+                    var packageSources = GetPackageSources(settings, sources, config);
+
+                    var frameworks = ParseArgumentWithCommas(framework.Value());
+                    
+                    var packageRefArgs = new ListPackageArgs(
+                        path.Value,
+                        packageSources,
+                        frameworks,
+                        includeOutdated.HasValue(),
+                        includeDeprecated.HasValue(),
+                        includeTransitive.HasValue(),
+                        prerelease.HasValue(),
+                        highestPatch.HasValue(),
+                        highestMinor.HasValue(),
+                        logger,
+                        CancellationToken.None
+                    );
+
+                    var listPackageCommandRunner = getCommandRunner();
+                    await listPackageCommandRunner.ExecuteCommandAsync(packageRefArgs);
+                    return 0;
+                });
+            });
+        }
+
+        private static ISettings ProcessConfigFile(string configFile, string currentDirectory)
+        {
+            if (string.IsNullOrEmpty(configFile))
+            {
+                return Settings.LoadDefaultSettings(currentDirectory);
+            }
+            else
+            {
+                var configFileFullPath = Path.GetFullPath(configFile);
+                var directory = Path.GetDirectoryName(configFileFullPath);
+                var configFileName = Path.GetFileName(configFileFullPath);
+                return Settings.LoadDefaultSettings(
+                    directory,
+                    configFileName,
+                    machineWideSettings: new XPlatMachineWideSetting());
+            }
+
+        }
+
+        private static List<PackageSource> GetPackageSources(ISettings settings, List<string> sources, CommandOption config)
+        {
+            
+            var sourceProvider = new PackageSourceProvider(settings);
+            var availableSources = sourceProvider.LoadPackageSources().Where(source => source.IsEnabled).ToList();
+            var packageSources = new List<PackageSource>();
+            foreach (var source in sources)
+            {
+                packageSources.Add(PackageSourceProviderExtensions.ResolveSource(availableSources, source));
+            }
+
+            if (packageSources.Count == 0 || config.HasValue())
+            {
+                packageSources.AddRange(availableSources);
+            }
+
+            return packageSources;
+        }
+
+        private static List<string> ParseArgumentWithCommas(string argument)
```

Comment:
as a rule of thumb, any method signature should always take in interfaces, and return type should also be an interface rather than a concrete implementation. please incorporate this everywhere in the code.

Reply:
On it